pkg/eosclientgrpc/eosclientgrpc.go

```diff
@@ -0,0 +1,1347 @@
+	if err != nil {
+		fmt.Printf("--- getACLForPath '%s' failed with err '%s'\n", "/eos/cms", err)
+		//	return nil
+	} else {
+		for i, s := range arep.Entries {
+			fmt.Printf("--- getACLForPath to '%s' gave %d:'%s'\n", "/eos/cms", i, s)
+		}
+	}
+
+	return c
+}
+
+func (c *Client) getUnixUser(username string) (*gouser.User, error) {
+	if c.opt.ForceSingleUserMode {
+		username = c.opt.SingleUsername
+	}
+	return gouser.Lookup(username)
+}
+
+// AddACL adds an new acl to EOS with the given aclType.
+func (c *Client) AddACL(ctx context.Context, username, path string, a *acl.Entry) error {
+	acls, err := c.getACLForPath(ctx, username, path)
+	if err != nil {
+		return err
+	}
+
+	// since EOS Citrine ACLs are is stored with uid, we need to convert username to uid
+	// only for users.
+	if a.Type == acl.TypeUser {
+		a.Qualifier, err = getUID(a.Qualifier)
+		if err != nil {
+			return err
+		}
+	}
+	err = acls.SetEntry(a.Type, a.Qualifier, a.Permissions)
+	if err != nil {
+		return err
+	}
+	sysACL := acls.Serialize()
+
+	// Stuff filename, uid, gid into the MDRequest type
+	rq := new(erpc.NSRequest)
+
+	// setting of the sys.acl is only possible from root user
+	unixUser, err := c.getUnixUser(username)
+	if err != nil {
+		return err
+	}
+	rq.Role = new(erpc.RoleId)
+
+	uid, err := strconv.ParseUint(unixUser.Uid, 10, 64)
+	if err != nil {
+		return err
+	}
+	rq.Role.Uid = uid
+	gid, err := strconv.ParseUint(unixUser.Gid, 10, 64)
+	if err != nil {
+		return err
+	}
+	rq.Role.Gid = gid
+
+	rq.Authkey = c.opt.Authkey
+
+	msg := new(erpc.NSRequest_AclRequest)
+	msg.Cmd = erpc.NSRequest_AclRequest_ACL_COMMAND(erpc.NSRequest_AclRequest_ACL_COMMAND_value["MODIFY"])
+	msg.Type = erpc.NSRequest_AclRequest_ACL_TYPE(erpc.NSRequest_AclRequest_ACL_TYPE_value["SYS_ACL"])
+	msg.Recursive = true
+	msg.Rule = sysACL
+
+	msg.Id = new(erpc.MDId)
+	msg.Id.Path = []byte(path)
+
+	rq.Command = &erpc.NSRequest_Acl{Acl: msg}
+
+	// Now send the req and see what happens
+	resp, err := c.cl.Exec(context.Background(), rq)
+	if err != nil {
+		fmt.Printf("--- Exec('%s') failed with err '%s'\n", path, err)
+		return err
+	}
+
+	fmt.Printf("--- MD('%s') gave response '%s'\n", path, resp)
+	if resp == nil {
+		return errtypes.NotFound(fmt.Sprintf("PAth: %s", path))
+	}
+
+	return err
+
+}
+
+// RemoveACL removes the acl from EOS.
+func (c *Client) RemoveACL(ctx context.Context, username, path string, aclType string, recipient string) error {
+	acls, err := c.getACLForPath(ctx, username, path)
+	if err != nil {
+		return err
+	}
+
+	// since EOS Citrine ACLs are is stored with uid, we need to convert username to uid
+	// only for users.
+
+	// since EOS Citrine ACLs are stored with uid, we need to convert username to uid
+	if aclType == acl.TypeUser {
+		recipient, err = getUID(recipient)
+		if err != nil {
+			return err
+		}
+	}
+	acls.DeleteEntry(aclType, recipient)
+	sysACL := acls.Serialize()
+
+	// Stuff filename, uid, gid into the MDRequest type
```

Comment:
The part of `RemoveACL` starting from this line is the same as that of `AddACL`, so we can modularize that.